modoboa/admin/app_settings.py

```diff
@@ -173,6 +194,22 @@ def clean_default_mailbox_quota(self):
             )
         return self.cleaned_data["default_mailbox_quota"]
 
+    def clean_dkim_keys_storage_dir(self):
+        """Check that directory exists."""
+        storage_dir = self.cleaned_data.get("dkim_keys_storage_dir", "")
+        if storage_dir:
+            if not os.path.isdir(storage_dir):
+                raise forms.ValidationError(
+                    ugettext_lazy("Directory not found")
+                )
+            code, output = exec_cmd("which openssl")
```

Comment:
A pure python solution? `if not which("openssl"):`

```python
import os

def which(program):
    def is_exe(fpath):
        return os.path.isfile(fpath) and os.access(fpath, os.X_OK)

    fpath, fname = os.path.split(program)
    if fpath:
        if is_exe(program):
            return program
    else:
        for path in os.environ["PATH"].split(os.pathsep):
            exe_file = os.path.join(path, program)
            if is_exe(exe_file):
                return exe_file

    return None
```

Reply:
Yes, why not. But it would require to update every calls to ``which`` (not necessarily related to this PR) so I suggest we make a new PR. What do you think?

Reply:
Sure, I'll knock something up.